pkg/storage/unified/resource/server.go

```diff
@@ -552,23 +555,61 @@ func (s *server) Watch(req *WatchRequest, srv ResourceStore_WatchServer) error {
 		return err
 	}
 
-	// Start listening -- this will buffer any changes that happen while we backfill
+	// Start listening -- this will buffer any changes that happen while we backfill.
+	// If events are generated faster than we can process them, then some events will be dropped.
+	// TODO: Think of a way to allow the client to catch up.
 	stream, err := s.broadcaster.Subscribe(ctx)
 	if err != nil {
 		return err
 	}
 	defer s.broadcaster.Unsubscribe(stream)
 
-	since := req.Since
+	mostRecentRV := s.mostRecentRV.Load() // get the latest resource version
+	var initialEventsRV int64             // resource version coming from the initial events
 	if req.SendInitialEvents {
-		fmt.Printf("TODO... query\n")
-		// All initial events are CREATE
-
-		if req.AllowWatchBookmarks {
-			fmt.Printf("TODO... send bookmark\n")
+		// Backfill the stream by adding every existing entities.
+		// todo ? I don't think we need to send since here , ResourceVersion: since
+		initialEventsRV, err = s.backend.ListIterator(ctx, &ListRequest{Options: req.Options}, func(iter ListIterator) error {
```

Comment:
IIUC, if the request asks for `SendInitialEvents` and it also _also_ provides a resource version (in our request, the `Since` field), then we should send the first bookmark event immediately after the event with the requested resource version.
Without specifying `Since`, if `SendInitialEvents` is set then we should send the bookmark after we replayed history until now (as it is in this PR).

See: https://kubernetes.io/docs/reference/using-api/api-concepts/#streaming-lists